src/log-pcap.c

```diff
@@ -220,11 +230,32 @@ void PcapLogRegister(void)
     (prof).total += (UtilCpuGetTicks() - pcaplog_profile_ticks); \
     (prof).cnt++
 
-static int PcapLogCondition(ThreadVars *tv, const Packet *p)
+static int PcapLogCondition(ThreadVars *tv, void *thread_data, const Packet *p)
 {
+    PcapLogThreadData *ptd = (PcapLogThreadData *)thread_data;
+
+    /* Log alerted flow or tagged flow */
+    switch (ptd->pcap_log->conditional) {
+        case LOGMODE_COND_ALERTS:
+            if (p->alerts.cnt || (p->flow && FlowHasAlerts(p->flow))) {
```

Comment:
It could be, but isn't necessary. The PKT_FIRST_ALERTS flag is used to determine when the data held in the tcp segments should be dumped (i.e. before the first alerted packet is dumped, otherwise will be out of order). The FlowSetHasAlertsFlag is set in the same block as p->flags |= PKT_FIRST_ALERTS.